getting-started/git-boot-camp.rst

```diff
@@ -669,3 +669,65 @@ Examples of useful commands:
 * Set the browser::
 
       $ gh config set browser <browser-path>
+
+
+Git worktree
+------------
+
+When working on several version branches at once, experienced Git users can
+take advantage of Git's worktree to check out several branches at once.
+
+Setting up Git worktree
+^^^^^^^^^^^^^^^^^^^^^^^
+
+With an existing CPython clone (see :ref:`clone-your-fork`), rename the
+``cpython`` directory to ``main`` and move it into a new ``cpython``
+directory, so we have a structure like:
+
+.. Generated with: tree -L 1 -d cpython
+
+.. code-block:: text
+
+   cpython
+   └── main (.git is here)
+
+Next, create worktrees for the other branches::
+
+   $ cd cpython/main
+   $ git worktree add -b 3.11 ../3.11 upstream/3.11
+   $ git worktree add -b 3.12 ../3.12 upstream/3.12
+
+This gives a structure like this, with the code for each branch checked out in
+its own directory:
+
+.. code-block:: text
+
+   cpython
+   ├── 3.11
+   ├── 3.12
+   └── main
+
+Using Git worktree
+^^^^^^^^^^^^^^^^^^
+
+List your worktrees, for example::
+
+   $ git worktree list
+   /Users/my-name/cpython/main  b3d24c40df [main]
+   /Users/my-name/cpython/3.11  da1736b06a [3.11]
+   /Users/my-name/cpython/3.12  cf29a2f25e [3.12]
+
+Change into a directory to work from that branch. For example::
+
+   $ cd ../3.12
+   $ git checkout my-3.12-bugfix-branch
+   ...
+   $ cd ../main
+   $ git checkout my-feature-branch
```

Comment:
Maybe this example would be more realistic:
```suggestion
   $ cd ../3.12
   $ git switch -c my-3.12-bugfix-branch
   ...
   $ cd ../main
   $ git switch -c my-feature-branch
```

I also assume that at the end one will have to remember to switch back to `3.12` and not `main` (in the `/3.12` dir), right?  Do we need to show a more complete example like:
```suggestion
   $ cd ../3.12
   $ git switch -c my-3.12-bugfix-branch  # create new branch
   $ # make changes, test them, commit
   $ git push origin my-3.12-bugfix-branch
   $ # create PR
   $ git switch 3.12  # switch back to the 3.12 branch
 ```
?

Anything else that I'm missing for this workflow?

Reply:
Yeah, I should have used `checkout -b` instead of `git checkout`. I use `checkout` because my fingers have learned it, but I guess `switch` is the better recommendation.

And yes, that looks good to switch back to `3.12` at the end. Let's go with your second suggestion. Thanks!